api/base/serializers.py

```diff
@@ -769,6 +775,25 @@ def invalid_embeds(self, fields, embeds):
         invalid_embeds = set(embeds.keys()) - set([f.field_name for f in fields_check if getattr(f, 'json_api_link', False)])
         return invalid_embeds
 
+    def to_esi_representation(self, data):
+        href = None
+        query_params_blacklist = ['page[size]', 'format']
+        try:
+            href = data.get_absolute_url()
+        except:
+            representation = super(JSONAPISerializer, self).to_representation(data)
+            return representation
+        else:
+            if href and not href == '{}':
```

Comment:
@brianjgeiger You're right. I stole that from a previous implementation and was returning a SkipField in the else clause. I just added in a failsafe. I should also probably make that exception clause be tighter. I'll figure out what to watch for an change that too.